Fix minimum JDK requirements (#10359)
infinispan/infinispan: documentation/src/main/asciidoc/topics/contributing/basics.adoc

== Pre-requisites
|=============

> | Java 11   | {brandname} is baselined on Java 11 but needs to be built with Java 11 (we build with OpenJDK)

Reviewer: ```suggestion
| Java 11   | {brandname} is built using and requires Java 11 for runtime (we build with OpenJDK)
```
Or something similar. The wording with baselined and built with above sounds a bit weird with the same version.
